lib/fuse_lowlevel.c

```diff
@@ -186,8 +186,13 @@ static int fuse_send_msg(struct fuse_session *se, struct fuse_chan *ch,
 		}
 	}
 
-	ssize_t res = writev(ch ? ch->fd : se->fd,
-			     iov, count);
+	ssize_t res;
+	if (se->io != NULL)
```

Comment:
Can we instead test for `se->io != NULL && se->io->writev !+ NULL` to rule out a potential bug and to increase flexibility?

Reply:
I considered that option as well, and if you'd prefer it, I wouldn't mind changing it. I'll give you the reason I omitted and let you decide the way forward

Given the condition in `fuse_session_custom_io`, which requires `writev` to be implemented, there is no way of `se->io->writev` being `NULL` if `se->io != NULL`. This is unless the user starts the session loop even though `fuse_session_custom_io` did not return 0, which would be a bug. So essentially I omitted the `se->io->writev != NULL` to indicate to the next person reading the code that there is no way of `se->io->writev` being `NULL` if  `se->io != NULL`.

Reply:
Ok, then let's just add a comment here that says that fuse_session_custom_io guarantees that if io != NULL, we also have io->writev != NULL.

Reply:
The same semantics is used for all the io operation calls (line 820, 2814 and 2907), should I comment it at all places?

Reply:
All places would be best, I think.